openedx/features/pakx/lms/overrides/views.py

```diff
@@ -48,6 +50,55 @@
 from xmodule.modulestore.django import modulestore
 
 
+# NOTE: This view is not linked to directly--it is called from
+# branding/views.py:index(), which is cached for anonymous users.
+# This means that it should always return the same thing for anon
+# users. (in particular, no switching based on query params allowed)
+def index(request, extra_context=None, user=AnonymousUser()):
+    """
+    Render the edX main page.
+
+    extra_context is used to allow immediate display of certain modal windows, eg signup.
+    """
+    if extra_context is None:
+        extra_context = {}
+    featured_course = get_featured_course()
+
+    courses_for_user = get_courses_for_user(user)
+
+    context = {'courses': courses_for_user, 'featured_course': featured_course,
+               'homepage_overlay_html': configuration_helpers.get_value('homepage_overlay_html'),
+               'show_partners': configuration_helpers.get_value('show_partners', True),
+               'show_homepage_promo_video': configuration_helpers.get_value('show_homepage_promo_video', False),
+               'homepage_course_max': configuration_helpers.get_value(
+                   'HOMEPAGE_COURSE_MAX', settings.HOMEPAGE_COURSE_MAX
+               )}
+
+    # This appears to be an unused context parameter, at least for the master templates...
+
+    # TO DISPLAY A YOUTUBE WELCOME VIDEO
+    # 1) Change False to True
+
+    # Maximum number of courses to display on the homepage.
+
+    # 2) Add your video's YouTube ID (11 chars, eg "123456789xX"), or specify via site configuration
+    # Note: This value should be moved into a configuration setting and plumbed-through to the
+    # context via the site configuration workflow, versus living here
+    youtube_video_id = configuration_helpers.get_value('homepage_promo_video_youtube_id', "your-youtube-id")
```

Comment:
I think we should set None instead of `your-youtube-id` like if no ID exists we should hide that section as well?